### PR TITLE
feat(task-worker): Add task worker consumer

### PR DESCRIPTION
**Pull request by enochtangg**
### Overview
Adds a python taskworker consumer and unit test. This PR ports over the consumer code from this [branch](https://github.com/getsentry/sentry/pull/76394/files#diff-8ab93612f6d482522dfd1b585975ff0e44d81383795616280ff7d90ec07a94d3), and makes some changes around typing, file naming, and lifting CLI args. 

Note: This consumer is temporary as we plan to migrate the consumer into a separate rust project. The intention is to lay a foundation to iterate on, make testing other components (task registry, namespace, GRPC, and workers) faster, and gather a round of feedback before implementing in rust.

### Depends on
* Merge https://github.com/getsentry/sentry/pull/79742 which contains new version of sentry-protos

### CODE REVIEW

File: src/sentry/taskworker/consumers/factory.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        self.inflight_task_store = InflightTaskStoreSqlite()
+
+    def create_with_partitions(
+        self, commit: Commit, _: Mapping[Partition, int]
+    ) -> ProcessingStrategy[KafkaPayload]:
+        def process_message(message: Message[KafkaPayload]) -> InflightActivation:
+            activation = TaskActivation()
+            activation.ParseFromString(message.payload.value)
+            ((_partition, offset),) = message.committable.items()
+            deadletter_at = int(time.time()) + self.max_pending_timeout
```
Comment (enochtangg) [inline src/sentry/taskworker/consumers/factory.py:62]: Does using `max_pending_timeout` make sense for calculating `deadletter_at` here?

File: src/sentry/consumers/__init__.py [hunk 2]
```diff
@@ -62,6 +62,37 @@ def multiprocessing_options(
...
+    options.append(
+        click.Option(
+            ["--max-inflight-activation-in-store", "max_inflight_activation_in_store"],
+            type=int,
+            default=1000,
+            help="Maximum number of pending inflight activations in the store before backpressure is emitted",
+        ),
+    )
+    if include_worker_addrs:
+        options.append(click.Option(["--worker-addrs"], type=str, default="127.0.0.1:50051"))
```
Comment (markstory) [inline src/sentry/consumers/__init__.py:91]: I don't think we're going to need this option as we're not going to have the consumer sending requests to the workers.

File: src/sentry/taskworker/consumers/factory.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            next_step=collect,
+        )
+
+        limit = RunTask(
+            function=limit_tasks,
+            next_step=process,
+        )
+
+        return RunTask(
+            function=do_upkeep,
```
Comment (evanh) [inline src/sentry/taskworker/consumers/factory.py:137]: This has the upkeep happening before the message is processed. Wouldn't we want to process the message, commit the offsets, and then do upkeep?

**Comment by markstory** [comment]
Going to close this as we'll be implementing the consumer in rust.

**Pull request closed without merging**